src/Validation/Factory.php

```diff
@@ -65,6 +65,15 @@ public function isInt() {
 		return new Constraints\IsInt($this->data_factory);
 	}
 
+	/**
+	 * Get a constraint for an array.
+	 *
+	 * @return  Constraint
+	 */
+	public function isArray() {
+		return new Constraints\IsArray($this->data_factory);
+	}
```

Comment:
IMO this constraint is super weak since arrays could take a lot of forms as the mashup dict-hashmap-list-thingy they are in PHP. Maybe we should have a look at this again when there is some implementation.

Reply:
I would suggest to introduce (and then of course use) `isArrayOf(Constraint $on_element)` instead. This defines the boundary of our system after all. By using this weak constraint bad people could slip an array of any content through this, which could then cause all kinds of havoc in the system.